src/runtime/PromiseObject.cpp

```diff
@@ -325,4 +326,69 @@ Value promiseAllResolveElementFunction(ExecutionState& state, Value thisValue, s
     }
     return Value();
 }
+
+Value promiseThenFinally(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    // Let onFinally be F.[[OnFinally]].
+    Value onFinally = argv[1];
```

Comment:
This should be retrieved by resolving the caller of the function, (which should be `PromiseStepFunctionObject` kind), then accessing it's `m_onFinally` member.